File: simple-jni/srcCopy/main/java/com/example/simplejni/HelloStachka.java
package com.example.simplejni;

/**
 * Copy of {@link HelloStachka.kt} to generate JNI headers via `javac -h`.
 * <p>
 * See https://youtrack.jetbrains.com/issue/KT-35127
 */
public final class HelloStachka {
    private static native String sayHello(String name);
}

File: simple-jni/src/main/java/com/example/simplejni/HelloStachka.kt
package com.example.simplejni

import com.example.libloader.LibLoader
import com.example.logging.simpleLogging

private const val TAG = "HelloStachka"

/**
 * Entry point to a simple JNI call.
 */
object HelloStachka {

    init {
        LibLoader.loadLibrary(BuildConfig.LIB_NAME)
    }

    external fun sayHello(name: String): String

    @JvmStatic
    private fun logCall() {
        simpleLogging(TAG) { "Called from native" }
    }
}
